intents/oversights/top_down_error.py

```diff
@@ -0,0 +1,247 @@
+
+    initial_row_i = 0
+    new_row_i = 0
+
+    suggestion_row_list = []
+
+    # dominant percentage is the percentage of pairs where 
+    # (first value - second value) is positive.
+
+    while initial_row_i < initial_num_rows:
+        if initial_row_i == initial_num_rows - 1 or initial_table_matrix[initial_row_i][:dimensions_len] != initial_table_matrix[initial_row_i + 1][:dimensions_len]:
+            while new_row_i < new_num_rows and initial_table_matrix[initial_row_i][:dimensions_len] == new_table_matrix[new_row_i][:dimensions_len]:
+                new_row_i = new_row_i + 1
+        else:
+            new_max_corelation = 0
+            while new_row_i < new_num_rows and initial_table_matrix[initial_row_i][:dimensions_len] == new_table_matrix[new_row_i][:dimensions_len]:
+                if new_row_i == new_row_i - 1 or new_table_matrix[new_row_i][:dimensions_len + 1] != new_table_matrix[new_row_i + 1][:dimensions_len + 1]:
+                    max_disimilarity = constants.TOP_DOWN_ERROR_DISSIMILARITY_THRESHOLD
+                else:
+                    corelation = _calculate_relation(new_table_matrix[new_row_i][dimensions_len + 2], new_table_matrix[new_row_i + 1][dimensions_len + 2])
+                    new_max_corelation = max(corelation, new_max_corelation)
+                    new_row_i = new_row_i + 1
+                new_row_i = new_row_i + 1
+
+            initial_corelation = _calculate_relation(initial_table_matrix[initial_row_i][dimensions_len + 1], initial_table_matrix[initial_row_i + 1][dimensions_len + 1])
+
+            if new_max_corelation >= constants.TOP_DOWN_ERROR_DISSIMILARITY_THRESHOLD and initial_corelation <= constants.TOP_DOWN_ERROR_SIMILARITY_THRESHOLD:
+                suggestion_row_list.append(initial_row_i)
+                suggestion_row_list.append(initial_row_i + 1)
+            initial_row_i = initial_row_i + 1
+
+        initial_row_i = initial_row_i + 1
+
+    if len(suggestion_row_list) == 0:
+        return None
+    else:
+        new_suggestion = {}
+        new_suggestion['suggestion'] = 'both the value is almost same in some grouping but different if we add ' + new_added_column + ' in initial grouping.'
+        new_suggestion['oversight_name'] = 'top-down error'
+        new_suggestion['is_row_level_suggestion'] = True
+        new_suggestion['row_list'] = suggestion_row_list
+        return new_suggestion
+
+def _calculate_relation(val1, val2):
+    """
+    This function can find the similarity between two values
+    """
+
+    """
+    Arg:
+        val1: the first value for which we have to compute the similarity
+        val2: the second value for which we have to compute the similarity
+
+    Returns:
+        return the similarity between both the arguments calculated by the
+        formula
+        similarity = |val1 - val2| / (|val1| + |val2|)
+    """
+
+    if val1 == val2:
+        return 0
+    result = abs(val1 - val2) / (abs(val1) + abs(val2))
```

Comment:
Duplication of code. A better check would be -
if (abs(val1) + abs(val2)) == 0 :
    return 0

Reply:
Done